packages/lib-react-components/src/Markdownz/Markdownz.js

```diff
@@ -1,34 +1,152 @@
+    let alt = nodeProps.alt
+    const src = nodeProps.src
+    const match = alt.match(imgSizeRegex)
+
+    if (match && match.length > 0) {
+      width = match[1]
+      height = match[3]
+      altText = alt.split(match[0])[0].trim()
+    }
+    const mimeType = mime.lookup(src)
+
+    if (mimeType && mimeType.includes('video')) {
+      return <Video a11yTitle={alt} controls="below" preload='metadata' src={src}><Paragraph>This video file is not supported</Paragraph></Video>
+    }
+    if (mimeType && mimeType.includes('audio')) {
+      // Grommet doesn't have an audio component right now
+      return <audio aria-title={alt} controls preload='metadata' src={src}><Paragraph>This audio file is not supported</Paragraph></audio>
+     }
+
+    return <Image alt={altText || alt} src={src} width={width} height={height} />
   }
-  const sanitizedMarkdownContent = sanitizeForMarkdown(children)
-
-  // Grommet Markdown is not styling anchors correctly to theme
-  // Issue: https://github.com/grommet/grommet/issues/2515
-  return (
-    <GrommetMarkdown options={mergedOptions}>
-      {sanitizedMarkdownContent}
-    </GrommetMarkdown>
-  )
+
+  render() {
+    const { children, components, settings } = this.props
+
+    if (!children) return null
+
+    if (Object.keys(components).includes('img')) {
+      console.warn('Overriding the rendering function for the img tag may break the syntax support for image resizing and using image markup for video and audio. Are you sure you want to do this?')
+    }
+
+    const componentMappings = {
+      a: Anchor,
+      h1: (nodeProps) => <Heading level='1'>{nodeProps.children}</Heading>,
+      h2: (nodeProps) => <Heading level='2'>{nodeProps.children}</Heading>,
+      h3: (nodeProps) => <Heading level='3'>{nodeProps.children}</Heading>,
+      h4: (nodeProps) => <Heading level='4'>{nodeProps.children}</Heading>,
+      h5: (nodeProps) => <Heading level='5'>{nodeProps.children}</Heading>,
+      h6: (nodeProps) => <Heading level='6'>{nodeProps.children}</Heading>,
+      img: (nodeProps) => this.renderMedia(nodeProps),
+      p: Paragraph,
+      span: Text,
+      table: Table,
+      tfoot: TableFooter,
+      thead: TableHeader,
+      tbody: TableBody,
+      td: TableCell,
+      tr: TableRow
+    }
+
+    const remarkReactComponents = Object.assign({},  componentMappings, components)
+    const remarkSettings = Object.assign({}, { footnotes: true }, settings)
+
+    const markdown = remark()
+      .data('settings', remarkSettings)
+      .use(emoji)
+      .use(remarkSubSuper)
+      .use(externalLinks)
+      .use(ping, {
+        ping: (resource, symbol) => this.shouldResourceBeLinkable(resource, symbol), // We could support passing in a prop to call a function here
+        pingSymbols: [at, hashtag, subjectSymbol],
+        resourceURL: (resource, symbol) => this.buildResourceURL(resource, symbol),
+        matchRegex: /\@([\w\-.]+)(?<!\.)|#([-\w\d]{3,40})|(\^S[0-9]+)/
```

Comment:
`\@([\w\-.]+)\b` might work better than the `(?<!\.)` negative lookbehind here. It will catch all word boundaries at the end of a username. 

https://regexr.com/ says `?<!` may not be supported in all  browsers, but doesn't say which browsers do support it, so `\b` might be safer for cross-browser support too.

Reply:
Burned again on trying to use something new. Firefox and Edge don't support it yet: http://kangax.github.io/compat-table/es2016plus/ Thanks for that catch.